exec/vector/src/main/codegen/templates/ComplexCopier.java

```diff
@@ -102,15 +103,15 @@ private static FieldWriter getMapWriterForReader(FieldReader reader, MapWriter w
       return (FieldWriter) writer.<#if name == "Int">integer<#else>${uncappedName}</#if>(name);
     <#elseif minor.class?contains("VarDecimal")>
     case ${name?upper_case}:
-      return (FieldWriter) writer.${uncappedName}(name, reader.getType().getScale(), reader.getType().getPrecision());
+      return (FieldWriter) writer.${uncappedName}(name, type.getScale(), type.getPrecision());
```

Comment:
I think correct order is to have precision + scale. Could you please create Jira for refactoring?